### DISCUSSION TITLE
Few changes to support mono

### DISCUSSION THREAD

**Discussion by moile666** [discord-net/Discord.Net#2690] [Ideas]
Hi everyone,

I wanted to use Discord.Net on my NAS, running on mono.
I saw it was not supported so I builded it myself.
I had to make 2 small changes in source to make it work : 

class DefaultRestClient :
```
//var content = new MultipartFormDataContent("Upload----" + DateTime.Now.ToString(CultureInfo.InvariantCulture));
var content = new MultipartFormDataContent("Upload----" + DateTimeOffset.UtcNow.ToUnixTimeSeconds());
```
because spaces are not supported in boundary argument.

class RequestQueue, method EnterGlobalAsync :
```
        internal async Task EnterGlobalAsync(int id, RestRequest request)
        {
            //int millis = (int)Math.Ceiling((_waitUntil - DateTimeOffset.UtcNow).TotalMilliseconds);
            double millis = Math.Ceiling((_waitUntil - DateTimeOffset.UtcNow).TotalMilliseconds);
            if (millis > 0)
            {
#if DEBUG_LIMITS
                Debug.WriteLine($"[{id}] Sleeping {millis} ms (Pre-emptive) [Global]");
#endif
                //await Task.Delay(millis).ConfigureAwait(false);
                await Task.Delay((int)millis).ConfigureAwait(false);
            }
        }
```

because, the cast makes the app wait for a long long time (like forever).

These changes are not breaking other stuff (I guess) and could probably be integrated for mono users.

Regards,

**Comment by Hawxy** [comment]
.NET Core/6/7 already runs on Linux, you shouldn't need to use Mono.
